tests/test_publish.py

```diff
@@ -85,56 +107,98 @@ def test_publish_float_2d(self, pin_query_context, published_data_reader):
             assert published_data_point.double_value == test_data_point
 
     def test_publish_bool_2d(self, pin_query_context, published_data_reader):
-        # [DigitalPattern1(ch4), DigitalPattern2(ch0)]
-        test_data = [[True], [False]]
+        # [[DigitalPattern1(ch6)], [DigitalPattern2(ch0), DigitalPattern2(ch1)]]
+        test_data = [[True], [False, True]]
         pin_query_context.publish(test_data)
         published_data = published_data_reader.get_and_clear_published_data()
         flattened_test_data = [data_point for row in test_data for data_point in row]
         assert len(published_data) == len(flattened_test_data)
         for published_data_point, test_data_point in zip(published_data, flattened_test_data):
             assert published_data_point.boolean_value == test_data_point
 
+    def test_publish_pattern(self, standalone_tsm_context, published_data_reader):
+        (
+            pin_query_context,
+            sessions,
+            site_lists,
+        ) = standalone_tsm_context.pins_to_nidigital_sessions_for_pattern(self._PIN)
+        expected_results = [True, False, True]  # test data across sites [0, 1, 2]
+        pattern = re.compile(r"\s*site(\d+)")
+        site_lists = [
+            [int(match[1]) for match in map(pattern.match, site_list.split(","))]
+            for site_list in site_lists
+        ]
+        test_data = [  # creates data for purely jagged results [[site0], [site1, site2]]
+            collections.OrderedDict((site, expected_results[site]) for site in site_list)
+            for site_list in site_lists
+        ]
```

Comment:
To make this easier to read, let's just use a simple definition: `test_data = [{0:True}, {2:True, 1:False}]`

Reply:
20c4a952b4a837227c4df69a06962da870e5d30b